### DISCUSSION TITLE
Confusion about get_text(1, 1, -1, -1)

### DISCUSSION THREAD

**Discussion by js-everts** [endaaman/tym#100] [General]
Its not clear to me from the documentation what this is supposed to do. Am I right in thinking if you do `get_text(a, b, -1, -1)`, then it supposed to return the contents of the terminal from `(a, b)` to `(a+rows, b+cols)` where `rows` and `cols` is the size of the terminal.

If so, lets say the terminal size is 80x20 with sufficient content in the scrollback buffer. If we do something like `get_text(100, 0, -1, -1)`, I think its supposed to be interpreted as `get_text(100, 0, 180, 20)`. But `tym` will return nothing because it will interpret this as `get_text(100, 0, 80, 20)` which is an invalid range. Is this by design or is this a bug?

Would it make sense to change `get_text(0, 0, -1, -1)` to return the entire scrollback buffer?. I know we can already do this by combining `get_cursor_position()`.

Also would it make sense to add a new builtin `get_text_visible()` to fetch the currently visible area of terminal. See `vte_terminal_get_text()`.

**Comment by endaaman** [comment]
`get_text()` API was added tentatively, I've been thinking it should be more refined. Please let me add to "My TODOs" a task to add these screen text controling APIs.
